### DISCUSSION TITLE
Using a Graph DB to store metadata

### DISCUSSION THREAD

**Discussion by jqnatividad** [ckan/ckan#6586] [Ideas]
Metadata naturally lends itself to graph databases, as it logically maps to how triples work, without having to deal with rigid metadata schemas.

In addition, it enables more natural, flexible, powerful queries without complex joins and even enables inference.

There have been several attempts to graft graph DBs to CKAN in the past, but they have not reached critical mass.

Now that we have the Apache AGE (A Graph Extension) project (https://age.apache.org/) that adds graph database functionality to PostgreSQL, perhaps, its time to revisit storing metadata as a graph and replace the `extras` tables?

Tabular data will still remain as regular SQL objects.  And it can all be done with PostgreSQL.  

Any thoughts?

**Comment by frafra** [comment]
Interesting idea, and thanks for sharing the AGE link.
That looks to me a concept that could be further developed in a CKAN extension, before being considered to be included in CKAN.

**Comment by ccancellieri** [comment]
Hi, yes, definitely it would be a great extension actually I'm missing the relationships a lot.

I'm planning to develop my own extension to use the graphdb capabilities to register and better maintain relations between packages (it would be a perfect fit with my jsonschema which allows also multiple strongly typed of resources).

Currently I'm still evaluating the platforms and the correct technologies to use:
f.e.: to avoid having an embedded neo4j installation in production, I'd like to use [Aura](https://neo4j.com/cloud/aura-google-cloud/)

Do you have some progress to share on this topic?
Best,
C.
